dashboard/src/shared/Secret.ts

```diff
@@ -0,0 +1,62 @@
+import axios from "axios";
+
+import { ISecret, IStatus } from "./types";
+
+export default class Secret {
+  public static async create(
+    name: string,
+    secrets: any,
+    owner: any | undefined,
+    namespace: string = "default",
+  ) {
+    const url = Secret.getLink(namespace);
+    try {
+      const { data } = await axios.post<ISecret>(url, {
+        apiVersion: "v1",
+        data: secrets,
+        kind: "Secret",
+        metadata: {
+          name,
+          ownerReferences: [Secret.getOwnerReference(owner)],
+        },
+        type: "Opaque",
+      });
+      return data;
+    } catch (e) {
+      throw new Error((e.response.data as IStatus).message);
+    }
+  }
+
+  public static async delete(name: string, namespace: string = "default") {
+    const url = this.getLink(namespace, name);
+    return axios.delete(url);
+  }
+
+  public static async get(name: string, namespace: string = "default") {
+    const url = this.getLink(namespace, name);
+    const { data } = await axios.get<ISecret>(url);
+    return data;
+  }
+
+  public static async list(namespace: string = "default") {
+    const url = Secret.getLink(namespace);
+    const { data } = await axios.get<ISecret>(url);
+    return data;
+  }
+
+  private static getOwnerReference(owner: any) {
+    return owner
+      ? {
```

Comment:
Let's add this as `IOwnerReference` to the types file instead of using `any`. We should always avoid using `any` unless there is a really good reason to do so.

Reply:
done!